src/adapter/src/catalog/open.rs

```diff
@@ -89,11 +88,11 @@ pub struct BuiltinMigrationMetadata {
         CatalogItemRebuilder,
     )>,
     pub introspection_source_index_updates:
-        BTreeMap<ClusterId, Vec<(LogVariant, String, GlobalId)>>,
+        BTreeMap<ClusterId, Vec<(LogVariant, String, GlobalId, u32)>>,
```

Comment:
What do you think making an `Oid(u32)` newtype?

Reply:
We actually already have this newtype in `mz_repr` (https://github.com/MaterializeInc/materialize/blob/d99e31d0c046c701305f13be13706ad86e291363/src/repr/src/adt/system.rs#L19-L23). I thought about using that type, but this PR had gotten big enough and we already weren't using that type for OID allocation. I would be open to switching the `u32` usages to `Oid` in a followup PR.